Loop/View Controllers/SettingsTableViewController.swift

```diff
@@ -620,72 +619,78 @@ final class SettingsTableViewController: UITableViewController {
-            } else if wasTestingPumpManager, !isTestingPumpManager {
-                guard let previousTestingPumpDataDeletionSection = previousTestingPumpDataDeletionSection else {
-                    fatalError("Expected to have had testing pump data deletion section when testing pump was in use")
-                }
-                tableView.deleteSections([previousTestingPumpDataDeletionSection], with: .automatic)
-            }
-            tableView.reloadSections([Section.pump.rawValue], with: .fade)
-            tableView.reloadSections([Section.cgm.rawValue], with: .fade)
-        case .cgm:
-            let previousTestingCGMDataDeletionSection = sections.firstIndex(of: .testingCGMDataDeletion)
-            let wasTestingCGMManager = isTestingCGMManager
-            isTestingCGMManager = dataManager.cgmManager is TestingCGMManager
-            if !wasTestingCGMManager, isTestingCGMManager {
-                guard let testingCGMDataDeletionSection = sections.firstIndex(of: .testingCGMDataDeletion) else {
-                    fatalError("Expected to find testing CGM data deletion section with testing CGM in use")
-                }
-                tableView.insertSections([testingCGMDataDeletionSection], with: .automatic)
-            } else if wasTestingCGMManager, !isTestingCGMManager {
-                guard let previousTestingCGMDataDeletionSection = previousTestingCGMDataDeletionSection else {
-                    fatalError("Expected to have had testing CGM data deletion section when testing CGM was in use")
-                }
-                tableView.deleteSections([previousTestingCGMDataDeletionSection], with: .automatic)
-            }
-            tableView.reloadRows(at: [indexPath], with: .fade)
-        case .configuration:
-            break
-        case .services:
-            break
-        case .testingPumpDataDeletion, .testingCGMDataDeletion:
-            break
-        }
-
-        return indexPath
-    }
-
     @objc private func dosingEnabledChanged(_ sender: UISwitch) {
         dataManager.loopManager.settings.dosingEnabled = sender.isOn
     }
 }
 
+// MARK: - DeviceManager view controller delegation
+
 extension SettingsTableViewController: CompletionDelegate {
     func completionNotifyingDidComplete(_ object: CompletionNotifying) {
-        if let vc = object as? UIViewController {
-            vc.dismiss(animated: true, completion: nil)
+        if let vc = object as? UIViewController, presentedViewController === vc {
+            dismiss(animated: true, completion: nil)
+
+            updateSelectedDeviceManagerRows()
         }
     }
+
+    private func updateSelectedDeviceManagerRows() {
+        updatePumpManagerRows()
+        updateCGMManagerRows()
+    }
+
+    private func updatePumpManagerRows() {
+        tableView.beginUpdates()
+
+        let previousTestingPumpDataDeletionSection = sections.firstIndex(of: .testingPumpDataDeletion)
```

Comment:
Yes we could, but we chose to bring this up to the Loop layer, to keep device managers free of dose store/hk concerns, especially if it would just be for non-production, testing code.